chroot_setup.py

```diff
@@ -0,0 +1,373 @@
+    os.environ['PATH'] = PATH
+    os.environ['LIBPATH'] = LIBPATH
+    print("**********************")
+    print("Live IBM i session (changes made).")
+    print("**********************")
+
+    if (ags.verbose):
+      print('PATH=' + PATH)
+      print('LIBPATH=' + LIBPATH)
+
+  else:
+    CHROOT_DEBUG = True
+    print("**********************")
+    print("Not IBM i, no action is taken (debug flow purpose only).")
+    print("**********************")
+    # exit -1
+
+
+def yum_install(package_list: list):
+  os.environ["LIBPATH"] = '/QOpenSys/pkgs/lib:/QOpenSys/usr/lib'
+  install_template = Template('/QOpenSys/pkgs/bin/yum $yes --installroot=$chroot_dir install $package')
+  auto_yes = '-y' if args.auto_yes else ''
+  for package in package_list:
+    command = install_template.substitute(yes=auto_yes, chroot_dir=args.chroot_directory, package=package)
+    logger.debug(command)
+    os.system(command)
+
+#
+# MAIN
+#
+
+lst = 0
+ops = 0
+qsys = 0
+CHROOT_DIR = ""
+CHROOT_LIST = ""
+SCRIPT_DIR = os.path.dirname(os.readlink(args.chroot_directory)) + "/config"
+
+# Didn't implement all of the checks, argparse can do that
+
+intial_check()
+
+# This is done because the arg may be a relative path name.
+# Need to be done before any 'cd'
+TEMP_DIR = os.readlink(args.chroot_directory)
+
+logger.debug("Scipt Dir: " + SCRIPT_DIR)
+os.chdir(SCRIPT_DIR)
+logger.debug("PWD: " + os.getcwd())
+
+if not TEMP_DIR.startswith('/QOpenSys/'):
+  print("\nERROR: [CHROOT DIRECTORY] Must begin with /QOpenSys/...\n")
+  parser.print_help
+  sys.exit(-2)
+
+CHROOT_DIR = TEMP_DIR
+
+
+if not os.path.isdir(CHROOT_DIR):
+  print(CHROOT_DIR + " does not exist.")
+  os.mkdir(CHROOT_DIR)
+  # TODO: Check creation
+else:
+  if not args.auto_yes
+    print(CHROOT_DIR + " directory already exists")
+    val = input("Would you like to continue chroot setup? [y/N]")
+    if not val == "y":
+      print("Bye")
+      sys.exit(-4)
+  else:
+    # continues automatically
+
+
+## CHANGE
+# Validate CHROOT TYPE Argument(s)
+if not args.chroot_type:
+  # Only chroot_directory was provided to the script:
+  # Ask if minimal with includes chroot is desired
+  if not args.auto_yes:
+    val = input("Would you like to create a minimal chroot w/ includes @ " + CHROOT_DIR + "? [y/N]:")
+    if not val == "y":
+      print("Specify your desired [CHROOT TYPE]")
+      parser.print_help
+      sys.exit(-5)
+
+  mylist = []
+  mylist.append(SCRIPT_DIR + "/chroot_includes.lst")
+  mylist.append(SCRIPT_DIR + "/chroot_minimal.lst")
+
+else:
+  # If chroot types were specified, then use those
+  for arg in args.chroot_type
+    print("Chroot type is: " + arg)
+    # TODO: I think this is line with the shell script, but what about random
+    # list names that might include 'ops' in them?
+    if ("ops" in arg):
+
+      # verify a deprecated OPS file was not specified
+      print("OPS has been depricated, and replaced with YUM Packages. Run $0 with a valid [CHROOT TYPE]")
```

Comment:
We no longer even have the OPS lsts in our `config` dir so checking this check is probably obsolete and could be removed.

Reply:
Agreed, remove the check